api/src/main/java/net/kyori/adventure/audience/Audience.java

```diff
@@ -133,6 +135,34 @@
     return Audiences.COLLECTOR;
   }
 
+  /**
+   * Filters this audience.
+   *
+   * <p>The returned {@code Audience} may be the same, or a completely different one.</p>
+   *
+   * @param filter the filter
+   * @since 4.8.0
+   */
+  default @NonNull Audience filter(final @NonNull Predicate<? super Audience> filter) {
+    if(filter.test(this)) {
+      return this;
+    }
+    return empty();
+  }
+
+  /**
+   * Executes an action against all audiences.
+   *
+   * <p>If you implement {@code Audience} and not {@link ForwardingAudience} in your own code, and your audience forwards to
+   * other audiences, then you <b>must</b> override this method and provide each audience to {@code action}.</p>
+   *
+   * @param action the action
+   * @since 4.8.0
+   */
+  default void foreach(final @NonNull Consumer<? super Audience> action) {
```

Comment:
Would it be better to call this `forEachAudience`?

Reply:
Personally, I'd prefer the more specific names. Less change for a clash and it also means the ugly `foreach` doesn't have to exist!

Reply:
I prefer `forEeachAudience` it's much clearer, having `foreach` and `forEach` in a class would get confusing super fast.
Having it be `foreach` may also mess with people who have their IDE autocomplete for them, as they may get the method they don't want, and not notice the capitalisation difference.